app/Main.hs

```diff
@@ -41,14 +41,22 @@ versionOpt = infoOption (showVersion version)
 updateGolden :: FilePath -> IO ()
 updateGolden dir = do
   putStrLn "Replacing golden with actual..."
-  goldenTests <- listDirectory dir
-  forM_ goldenTests (mvActualToGolden dir)
+  go dir
   putStrLn "Finish..."
+ where
+  go dir = do
+    entries <- listDirectory dir
+    forM_ entries $ \entry -> do
```

Comment:
@ChickenProp I think this always checks if `actual` or `equal` are directories. Let's filter those two after listing the directory. Sounds good?

Reply:
There's a risk there that `actual` and `golden` could be path components in a test, and that would work fine as long as you don't have tests named `foo/actual[/...]` or `foo/golden[/...]` along with `foo`. If we filter those out here, then those tests would still work in a specfile, but hgold wouldn't be able to update them.

My inclination would be not to filter them out, but I'll defer to your judgement on the matter - if you think we should, I'm happy to make that change.

Reply:
Make sense! Let's keep it like that then. Thanks!